pkg/client/cache/listers.go

```diff
@@ -111,6 +111,44 @@ func (s storePodsNamespacer) List(selector labels.Selector) (api.PodList, error)
 	return pods, nil
 }
 
+// Please note that selector is filtering among the pods that have gotten into
+// the store; there may have been some filtering that already happened before
+// that.
+func (s storePodsNamespacer) ListWithPredicate(pickPod func(*api.Pod) bool) (api.PodList, error) {
```

Comment:
Even better, make predicate take an interface and then test for interfaces
that expose information useful to the indexer (i.e. Go code for naive
indexes and an optimized version used by the indexer for inversion).

On Jul 7, 2016, at 2:04 PM, David Eads <notifications@github.com> wrote:

In pkg/client/cache/listers.go
<https://github.com/kubernetes/kubernetes/pull/27600#discussion_r69955809>:

> @@ -111,6 +111,44 @@ func (s storePodsNamespacer) List(selector labels.Selector) (api.PodList, error)
>  	return pods, nil
>  }
>
> +// Please note that selector is filtering among the pods that have gotten into
> +// the store; there may have been some filtering that already happened before
> +// that.
> +func (s storePodsNamespacer) ListWithPredicate(pickPod func(*api.Pod) bool) (api.PodList, error) {

Do we already have any indexer based on the label yet? I search in
pkg/controllers, it seems we only index namespace or name.

Not yet, its on my list. I'm still not seeing why I can't have

func (s storePodsNamespacer) List(selector labels.Selector) (pods
api.PodList, err error) {
    return ListWithPredicate(func(pod *api.Pod) bool{
        return selector.Matches(labels.Set(pod.Labels))
    }
}

—
You are receiving this because you are on a team that was mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/kubernetes/kubernetes/pull/27600/files/eb66453972b391aef4217f46594ca6a34e3d2cbf#r69955809>,
or mute the thread
<https://github.com/notifications/unsubscribe/ABG_p_0NngxM6_ar9LM8oMm32q0Fmp8jks5qTT-agaJpZM4I4CdG>
.